filter/ArticleMedraXmlFilter.inc.php

```diff
@@ -192,6 +192,41 @@ function createArticleNode($doc, $pubObject) {
 			$this->appendTextMiningCollectionNodes($doc, $articleNode, $article, $submissionGalleys);
 		}
 
+		$issueId = $article->getCurrentPublication()->getData('issueId');
+		if ($cache->isCached('issues', $issueId)) {
+		    $issue = $cache->get('issues', $issueId);
+		} else {
+		    $issueDao = DAORegistry::getDAO('IssueDAO'); /* @var $issueDao IssueDAO */
+		    $issue = $issueDao->getById($issueId, $context->getId());
+		    if ($issue) $cache->add($issue, null);
+		}
+		$journalId = $issue->getData('journalId');
+		if(!empty($journalId)){
+		    $accessRights = null;
+		    $journalDao = DAORegistry::getDAO('JournalDAO'); /** @var JournalDAO $journalDao */
+		    $journal = $journalDao->getById($journalId);
+		    if($journal->getData('publishingMode') == 0){
```

Comment:
Maybe to use the constants instead of numbers, here and below, so that we know where they are used, in case we need to search for them.
Here PUBLISHING_MODE_OPEN.

Reply:
I defined some constants in O4DOIXmlFilter:

define('O4DOI_PUBLISHING_MODE_OPEN', '0');
define('O4DOI_PUBLISHING_MODE_SUBSCRIPTION', '1');
define('O4DOI_ISSUE_MODE_OPEN', '1');
define('O4DOI_ISSUE_MODE_SUBSCRIPTION', '2');
define('O4DOI_ISSUE_ARTICLE_MODE_OPEN', '1');

Hope this will be enough. Please let's know.
With regards,
Jehu Njikonga

Reply:
There are already constants in the core for this. In the Journal class.